### ISSUE TITLE
false positive for Lint/UselessAssignment with until loop

### ISSUE THREAD

**Issue by seandilda** [rubocop/rubocop#14402]
I have some text parsing code that passed all lints with rubocop 1.77.0 but has a Lint/UselessAssignment offense as of 1.79.0

I managed to simplify the code to this example:

```ruby
# frozen_string_literal: true

list = '3234(234)23'.chars
in_parens = false
until list.empty?
  c = list.shift
  if in_parens
    if c == ')'
      in_parens = false
    else
      $stdout.puts c
    end
  elsif c == '('
    in_parens = true
  end
end
```

--------

## Expected behavior

No offenses

## Actual behavior
```
$ rubocop --debug test.rb
For /app: configuration from /app/.rubocop.yml
Default configuration from /home/docker-user/bundle/ruby/3.3.0/gems/rubocop-1.79.0/config/default.yml
Plugin configuration from /home/docker-user/bundle/ruby/3.3.0/gems/rubocop-rails-2.32.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /app/test.rb
W

Offenses:

test.rb:9:7: W: [Correctable] Lint/UselessAssignment: Useless assignment to variable - in_parens.
      in_parens = false
      ^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.26479940899298526 seconds
```

## Steps to reproduce the problem

See sample code above

## RuboCop version

```
1.79.0 (using Parser 3.3.9.0, rubocop-ast 1.46.0, analyzing as Ruby 2.7, running on ruby 3.3.7) [x86_64-linux]
  - rubocop-rails 2.32.0
```

**Comment by bmulholland** [comment]
Probably dupe of https://github.com/rubocop/rubocop/issues/14383

**Comment by seandilda** [comment]
Agreed.  Sorry about the extra bug report.

**Comment by koic** [comment]
Unfortunately, this issue was not resolved by #14383. I’ll take a look at it later.

**Comment by kddnewton** [comment]
Same issue here: https://github.com/ruby-syntax-tree/syntax_tree/actions/runs/16503908321/job/46669472288?pr=528.

**Comment by toy** [comment]
Same here https://github.com/toy/image_size/actions/runs/16614072741/job/47003008168

**Comment by franzliedke** [comment]
I also still get a false positive for this code, despite #14383 being merged:

```ruby
index = 0
covered_periods = init_covered_periods_array_somehow

while index < covered_periods.length
  lookup_period = some_code

  if lookup_period.nil?
    index += 1 # <-- false positive reported here
    next
  end

  covered_periods = derive_somehow(lookup_period)
  index = 0
end
```

**Comment by kddnewton** [comment]
Still happening unfortunately https://github.com/ruby-syntax-tree/syntax_tree/actions/runs/16655668361/job/47139912763?pr=529#step:4:14

**Comment by koic** [comment]
@kddnewton This patch hasn’t been released yet, so it’s not included in RuboCop 1.79.1. It looks like the issue is resolved on the latest master branch, but is there still any issue?
